revelatormodsystem.py

```diff
@@ -2,49 +2,67 @@
 import sys
 import os
 
-scriptextract = False
+script_extract_folder = ''
 basepath = ''
+mod_folder = ''
+
+
 def bytearrtomessage(barr):
-	result = ""	
+	result = ""
 	for b in barr:
 		result += hex(b) + ','
 	print result[:4]
 	return result[:-1]
+
+
 def on_message(message, data):
-	global script
-	global scriptextract
-	global basepath
 	print message
-	currscriptpath = os.path.join(basepath, 'ggmods', message['payload'] + '.ggscript');
-	if scriptextract and not os.path.isfile(os.path.join(basepath, 'originalscripts', message['payload'] + '.ggscript')):
-		origscript = open(os.path.join(basepath, 'originalscripts', message['payload'] + '.ggscript'), 'wb')
-		origscript.write(bytearray(data))	
-		origscript.close()
-	elif scriptextract:
-		print 'originalscripts/' + message['payload'] + '.ggscript already exists.  Not overwriting.'
+	script_name = message['payload'] + '.ggscript'

```

Comment:
Imo it's better to keep the name of the script in a single variable rather than recomputing the path every single time.